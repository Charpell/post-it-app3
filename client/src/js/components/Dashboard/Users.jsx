import React, {Component} from 'react'


/**
 * @description Displays the List of Users in a group
 * 
 * @param props 
 * 
 * @class Users
 * 
 * @extends {Component}
 */
export default class Users extends Component {
  render() {
    return ( 
        <li data-toggle="collapse" className="collapsed">
            <a href="#"><i className="fa fa-globe fa-lg">
              </i>&nbsp; {this.props.KeyName.userName}
            </a>
        </li>     

    )
  }
}