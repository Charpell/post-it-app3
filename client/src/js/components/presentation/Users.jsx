import React from 'react';


/**
 * @description Displays the List of Users in a group
 *
* @param  {object} props store data passed to the component
*
* @class Users
*
* @extends {Component}
*/
const Users = () => ({
  render() {
    return (
      <li data-toggle="collapse" className="collapsed">
        <a href="#"><i className="fa fa-globe fa-lg">
        </i>&nbsp; {this.props.keyName.userName}
        </a>
      </li>
    );
  }
});

export default Users;
