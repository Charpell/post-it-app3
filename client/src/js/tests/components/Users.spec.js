import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Users from '../../components/Dashboard/Users'


describe('Users component', () => {
  it('Users component should render as expected', () => {
    const tree = renderer.create(<Users />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});