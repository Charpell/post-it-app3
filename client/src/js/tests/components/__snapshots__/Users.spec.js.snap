// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Users component should render as expected 1`] = `
<li
  className="collapsed"
  data-toggle="collapse"
>
  <a
    href="#"
  >
    <i
      className="fa fa-globe fa-lg"
    />
    Â  
  </a>
</li>
`;
